components/script/dom/htmlinputelement.rs

```diff
@@ -442,53 +446,86 @@ impl HTMLInputElementMethods for HTMLInputElement {
+        }
+    }
 
     // https://html.spec.whatwg.org/multipage/#dom-input-maxlength
     make_int_getter!(MaxLength, "maxlength", DEFAULT_MAX_LENGTH);
 
     // https://html.spec.whatwg.org/multipage/#dom-input-maxlength
     make_limited_int_setter!(SetMaxLength, "maxlength", DEFAULT_MAX_LENGTH);
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-min
+    // https://html.spec.whatwg.org/multipage/#dom-input-min
     make_getter!(Min, "min");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-min
-    make_setter!(SetMin, "min");
+    // https://html.spec.whatwg.org/multipage/#dom-input-min
+    fn SetMin(&self, value: DOMString) {
+        let element = self.upcast::<Element>();
+        if let Ok(val) = attr::parse_double(&value) {
+            element.set_double_attribute(&atom!("min"), val);
+        } else if self.type_() == atom!("range") {
+            element.set_double_attribute(&atom!("min"), DEFAULT_MIN_FOR_RANGE);
+        }
+    }
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-multiple
+    // https://html.spec.whatwg.org/multipage/#dom-input-multiple
     make_bool_getter!(Multiple, "multiple");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-multiple
+    // https://html.spec.whatwg.org/multipage/#dom-input-multiple
     make_bool_setter!(SetMultiple, "multiple");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-pattern
+    // https://html.spec.whatwg.org/multipage/#dom-input-pattern
     make_getter!(Pattern, "pattern");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-pattern
+    // https://html.spec.whatwg.org/multipage/#dom-input-pattern
     make_setter!(SetPattern, "pattern");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-required
+    // https://html.spec.whatwg.org/multipage/#dom-input-required
     make_bool_getter!(Required, "required");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-required
+    // https://html.spec.whatwg.org/multipage/#dom-input-required
     make_bool_setter!(SetRequired, "required");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-src
-    make_getter!(Src, "src");
+    // https://html.spec.whatwg.org/multipage/#dom-input-src
+    make_url_getter!(Src, "src");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-src
-    make_setter!(SetSrc, "src");
+    // https://html.spec.whatwg.org/multipage/#dom-input-src
+    make_url_setter!(SetSrc, "src");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-step
+    // https://html.spec.whatwg.org/multipage/#dom-input-step
     make_getter!(Step, "step");
 
-    // https://html.spec.whatwg.org/multipage/#attr-input-step
-    make_setter!(SetStep, "step");
+    // https://html.spec.whatwg.org/multipage/#dom-input-step
+    fn SetStep(&self, value: DOMString) {
+        let element = self.upcast::<Element>();
+        if &*value.to_lowercase() == "any" {
```

Comment:
You could use [`AsciiExt::eq_ignore_ascii_case`](https://doc.rust-lang.org/std/ascii/trait.AsciiExt.html#tymethod.eq_ignore_ascii_case) if you wanted to avoid a heap allocation here